src/main/java/org/jenkinsci/plugins/github_branch_source/Endpoint.java

```diff
@@ -48,19 +56,85 @@
  * @author Stephen Connolly
  */
 public class Endpoint extends AbstractDescribableImpl<Endpoint> {
+    /**
+     * Common prefixes that we should remove when inferring a display name.
+     */
+    private static final String[] COMMON_PREFIX_HOSTNAMES = {
+            "git.",
+            "gitea.",
+            "gogs.",
+            "vcs.",
+            "scm.",
+            "source."
+    };
+
     private final String name;
     private final String apiUri;
 
+    /**
+     * Makes best effort to guess a "sensible" display name from the hostname in the apiUri.
+     *
+     * @param apiUri the apiUri.
+     * @return the display name or {@code null}
+     * @throws LinkageError if Guava changes their API that we have depended on.
+     */
+    @CheckForNull
+    /*package*/ static String inferDisplayName(@CheckForNull String apiUri) throws LinkageError {
+        if (apiUri == null) {
+            return apiUri;
+        }
+        String hostName;
+        try {
+            URI serverUri = new URI(apiUri);
+            hostName = serverUri.getHost();
+            if (hostName != null) {
+                // let's see if we can make this more "friendly"
+                InternetDomainName host = InternetDomainName.from(hostName);
+                if (host.hasPublicSuffix()) {
+                    String publicName = host.publicSuffix().name();
+                    hostName = StringUtils.removeEnd(StringUtils.removeEnd(host.name(), publicName), ".")
+                            .toLowerCase(Locale.ENGLISH);
```

Comment:
I am not sure if it is a good approach 🐜 . It may sound bizarre, but there are non-Latin domain names. E.g. I know a company which uses Cyrillic domains in the internal network. And they use Jenkins.... It seems to be a wrong presumption though it should work in 99.9% cases